From cdd18f1ee4fe2391e1fda1e0dca8558c687dd2a9 Mon Sep 17 00:00:00 2001
From: Ankit Agrawal <ankit.agarwal@intel.com>
Date: Fri, 15 Dec 2023 10:02:25 +0530
Subject: [PATCH] Fixed for Crash in Contact app during Monkey Test.

During monkey test run, observing following crash
java.lang.IllegalStateException: Can not perform this action after
onSaveInstanceState at android.app.FragmentManagerImpl.checkStateLoss
(FragmentManager.java:1882)
at android.app.FragmentManagerImpl.enqueueAction(FragmentManager.java
:1905)
at android.app.BackStackRecord.commitInternal(BackStackRecord.java:688)
at android.app.BackStackRecord.commit(BackStackRecord.java:646)
at com.android.contacts.activities.PeopleActivity
.updateViewConfiguration(PeopleActivity.java:739)

when you use commit() it will can throw an exception if state loss
occurs but commitAllowingStateLoss() saves transaction without state
loss so that will doesn't throw an exception if state loss occurs.

Tests:
Run monkey test and observe, test is pass.

Tracked-On: OAM-113422
Signed-off-by: Ankit Agrawal <ankit.agarwal@intel.com>
---
 src/com/android/contacts/activities/PeopleActivity.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/com/android/contacts/activities/PeopleActivity.java b/src/com/android/contacts/activities/PeopleActivity.java
index 243ec003b..112fa1b93 100644
--- a/src/com/android/contacts/activities/PeopleActivity.java
+++ b/src/com/android/contacts/activities/PeopleActivity.java
@@ -736,7 +736,7 @@ public class PeopleActivity extends AppCompatContactsActivity implements
             fragment.updateStatus(mProviderStatus);
         }
         if (!transaction.isEmpty()) {
-            transaction.commit();
+            transaction.commitAllowingStateLoss();
             fragmentManager.executePendingTransactions();
         }
 
-- 
2.17.1

